nixos/modules/services/networking/sabnzbd/default.nix

```
@@ -0,0 +1,609 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkOption
+    mkPackageOption
+    mkEnableOption
+    mkOptionDefault
+    mkIf
+    literalExpression
+    types
+    ;
+  inherit (lib.generators)
+    mkKeyValueDefault
+    mkValueStringDefault
+    toKeyValue
+    ;
+  inherit (pkgs) sabnzbd;
+  cfg = config.services.sabnzbd;
+
+  mandatoryGlobalSettings = {
+    "__version__" = 19;
+    "__encoding__" = "utf-8";
+  };
+  allSettings = cfg.settings // mandatoryGlobalSettings;
+
+  # sabnzbd uses configobj type inis, which support
+  # nested sections specified by increasing numbers
+  # of square brackets (but not toml style dotted paths)
+  configObjAtom = types.oneOf [
+    types.str
+    types.int
+    types.bool
+  ];
+  configObjValue =
+    let
+      valueType =
+        types.oneOf [
+          types.str
+          types.int
+          types.bool
+          (types.listOf configObjAtom)
+          (types.attrsOf valueType)
+        ]
+        // {
+          description = "ConfigObj type";
+        };
+    in
+    valueType;
+  configObjIni =
+    { }:
+    let
+      extractAtoms = lib.filterAttrs (k: v: v != null && !lib.isAttrs v);
+      extractSections = lib.filterAttrs (k: v: lib.isAttrs v);
+      mkValueString = (
+        v:
+        if true == v then
+          "1"
+        else if false == v then
+          "0"
+        else
+          mkValueStringDefault { } v
+      );
+      mkKeyValue = mkKeyValueDefault { inherit mkValueString; } "=";
+      mkSection = (
+        depth: attrs:
+        let
+          atoms = extractAtoms attrs;
+          sections = extractSections attrs;
+          sectionHeadingLeft = lib.concatStrings (lib.replicate (depth + 1) "[");
+          sectionHeadingRight = lib.concatStrings (lib.replicate (depth + 1) "]");
+          mkSectionHeading =
+            name: "${sectionHeadingLeft}${lib.escape [ "[" "]" ] name}${sectionHeadingRight}";
+          mkSubsection = name: attrs: (mkSectionHeading name) + "\n" + (mkSection (depth + 1) attrs) + "\n";
+        in
+        toKeyValue { inherit mkKeyValue; } (extractAtoms attrs)
+        + "\n"
+        + lib.concatStrings (lib.mapAttrsToList mkSubsection sections)
+      );
+    in
+    {
+      type = types.attrsOf configObjValue;
+      generate = name: attrs: pkgs.writeText name (mkSection 0 attrs);
+    };
+  publicIni =
+    if cfg.configFile != null then
+      cfg.configFile
+    else
+      (configObjIni { }).generate "public-settings.ini" allSettings;
+  totalIni = "/var/lib/${cfg.stateDir}/sabnzbd.ini";
+in
+
+{
+
+  ###### interface
+
+  options = {
```

```suggestion
{
  options = {
```
those are legacy comments